cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/TextOutputFormatter.cs

```diff
@@ -0,0 +1,48 @@
+using System.CommandLine;
+using System.Text.Json;
+using Spectre.Console;
+
+namespace Microsoft.Kiota.Cli.Commons.IO;
+
+/// <summary>
+/// The JSON output formatter
+/// </summary>
+public class TextOutputFormatter : IOutputFormatter
+{
+    private readonly IAnsiConsole _ansiConsole;
+
+    /// <summary>
+    /// Creates a new JSON output formatter with a default console
+    /// </summary>
+    public TextOutputFormatter() : this(AnsiConsole.Console)
+    {
+    }
+
+    /// <summary>
+    /// Creates a new JSON output formatter with the provided console
+    /// </summary>
+    /// <param name="console">The console to use</param>
+    public TextOutputFormatter(IAnsiConsole console)
+    {
+        this._ansiConsole = console;
+    }
+
+    /// <inheritdoc />
+    public async Task WriteOutputAsync(Stream content, IOutputFormatterOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        using var reader = new StreamReader(content);
```

Comment:
why not using a string reader instead to handle the buffering for you?
https://docs.microsoft.com/en-us/dotnet/api/system.io.stringreader?view=net-6.0

Reply:
ignore this, I got confused.